### PR TITLE
Scale & Axis Migration MVP

### PR DESCRIPTION
**Pull request by kanitw**
Replace existing ordinal scale with the new 3 discrete scale types
- remove `points` from Scale interface

### CODE REVIEW

File: test/compile/mark/rect.test.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ describe('Mark: Rect', function() {
   describe('simple vertical', function() {
     const model = parseUnitModel({
       "data": {"url": 'data/cars.json'},
-      "mark": "bar",
+      "mark": "rect",
```
Comment (domoritz) [inline test/compile/mark/rect.test.ts]: should be in master, right?

File: package.json [hunk 2]
```diff
@@ -77,7 +77,7 @@
     "typescript": "^2.0.6",
     "typescript-json-schema": "~0.3.2",
     "uglify-js": "~2.7.4",
-    "vega": "~2.6.3",
+    "vega": "vega/vega",
```
Comment (domoritz) [inline package.json]: Can we use the beta version?

Comment (kanitw) [inline package.json]: We should but I don't see it being released yet..

**Comment by kanitw** [comment]
@domoritz  it's not 100% but I put FIXME in all skeptical ones.  I guess we should get something to render first / link editor with Vega 3 so we can debug easier.

**Comment by kanitw** [comment]
So I'll just for now to avoid to many branches

**Pull request merged**